File: check_markets.js
const https = require('https');

https.get('https://clob.polymarket.com/markets', (res) => {
    let data = '';
    res.on('data', (chunk) => data += chunk);
    res.on('end', () => {
        const markets = JSON.parse(data).data;
        
        console.log('Sample market 1:');
        const m = markets[0];
        console.log(JSON.stringify({
            question: m.question,
            active: m.active,
            closed: m.closed,
            archived: m.archived,
            end_date_iso: m.end_date_iso,
            tokens: m.tokens.length
        }, null, 2));
        
        console.log('\nSample market 50:');
        const m2 = markets[50];
        console.log(JSON.stringify({
            question: m2.question,
            active: m2.active,
            closed: m2.closed,
            end_date_iso: m2.end_date_iso
        }, null, 2));
        
        console.log('\nChecking first 100:');
        const activeCount = markets.slice(0, 100).filter(m => m.active).length;
        const notClosed = markets.slice(0, 100).filter(m => !m.closed).length;
        const notArchived = markets.slice(0, 100).filter(m => !m.archived).length;
        console.log('Active:', activeCount, '/100');
        console.log('Not closed:', notClosed, '/100');
        console.log('Not archived:', notArchived, '/100');
        
        // Find any from 2026
        console.log('\nSearching for 2026 markets...');
        const recent = markets.filter(m => {
            const q = m.question || '';
            return q.includes('2026');
        }).slice(0, 5);
        
        console.log('Found', recent.length, '2026 markets');
        recent.forEach((m, i) => {
            console.log(`${i + 1}. ${m.question.substring(0, 80)}`);
            console.log(`   Active: ${m.active}, Closed: ${m.closed}`);
            console.log(`   Token ID: ${m.tokens[0]?.token_id}`);
        });
    });
});

File: check_markets.py
import sqlite3
import json
from datetime import datetime

conn = sqlite3.connect('polymarket-monitor/polymarket_data.db')
cursor = conn.cursor()

# Get sample of markets
cursor.execute("SELECT category, COUNT(*) as count FROM markets GROUP BY category ORDER BY count DESC;")
categories = cursor.fetchall()
print("Market categories distribution:")
for cat, count in categories:
    print(f"  {cat}: {count}")

# Get date range of markets
cursor.execute("SELECT MIN(created_at), MAX(created_at) FROM markets;")
min_date, max_date = cursor.fetchone()
print(f"\nMarket creation date range: {min_date} to {max_date}")

# Get active vs inactive
cursor.execute("SELECT active, COUNT(*) FROM markets GROUP BY active;")
active_stats = cursor.fetchall()
print("\nActive status:")
for active, count in active_stats:
    print(f"  Active={active}: {count}")

# Get sample market questions
cursor.execute("SELECT question, category, created_at, end_date FROM markets LIMIT 10;")
samples = cursor.fetchall()
print("\nSample markets:")
for question, category, created, end in samples:
    print(f"  [{category}] {question[:80]}...")
    print(f"    Created: {created}, Ends: {end}")

# Check for resolution data in another way
print("\nChecking for any price/resolution data...")

# Look for any files with market data
import os
import glob

# Check for JSON files with market data
json_files = glob.glob("polymarket-monitor/**/*.json", recursive=True)
market_data_files = [f for f in json_files if 'market' in f.lower() or 'price' in f.lower()]
print(f"\nFound {len(market_data_files)} potential market data files")

conn.close()